src/vs/workbench/parts/terminal/electron-browser/terminalPanel.ts

```diff
@@ -101,11 +98,19 @@ export class TerminalPanel extends Panel {
 	) {
 		super(TERMINAL_PANEL_ID, telemetryService);
 		this.toDispose = [];
+		configurationService.onDidUpdateConfiguration(()=>{
+			this.applyFont();
```

Comment:
If #6456 gets done, this should reuse the the `onDidUpdateConfiguration` that's being merged into master soon https://github.com/Microsoft/vscode/pull/6985/commits/3fde03980baf7d33d14c36dce623ad1c89470d93